packages/nuxt/src/app/composables/router.ts

```diff
@@ -266,14 +266,19 @@ export const abortNavigation = (err?: string | Partial<NuxtError>) => {
   throw err
 }
 
-/** @since 3.0.0 */
-export const setPageLayout = (layout: unknown extends PageMeta['layout'] ? string : PageMeta['layout']) => {
+/**
+ * Sets the layout for the current page.
+ * TODO type props
+ * @since 3.0.0
+ */
+export const setPageLayout = <Layout extends keyof NuxtLayouts>(layout: unknown extends Layout ? string : Layout, props?: typeof layout extends Layout ? NuxtLayouts[Layout] : never) => {
```

Comment:
we probably do have to constrain the types to ensure they are serialisable (e.g. can't contain functions.....)